app/models/twitch_user.rb

```diff
@@ -39,9 +40,29 @@ def self.default_avatar
 
   def videos
     Twitch::Videos.recent(twitch_id, type: :archive, token: access_token)
+  rescue RestClient::Unauthorized
+    refresh_tokens!
```

Comment:
If we catch an error here and go to refresh the token, the original intended action is going to not be performed, correct?  Maybe we could check the return value of `refresh_tokens!` and then if it is successful call `retry` in the rescue block so that it process the original request for data?